### PR TITLE
WELD-2537, WELD-2539 Add WeldManager#isContextActive method. Add WeldManager#getScopes.

### PR DESCRIPTION
**Pull request by manovotn**
Depends on #1875 (for test) and on https://github.com/weld/api/pull/79 (for API).

### CODE REVIEW

**Comment by mkouba** [comment]
Shouldn't we test `getActiveContexts()` and `getActiveWeldAlterableContexts()` as well?

**Comment by manovotn** [comment]
I was planning on using those two as part of the context propagation tests.
But if you think it better, we can have some basic coverage here as well.

**Comment by manovotn** [comment]
Added the tests for active [weld alterable] contexts.

File: environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/weldManager/scopes/GetScopesTests.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    public void testGettingAllScopesInApplication() {
+        try (WeldContainer container = new Weld().initialize()) {
+            WeldManager wm = container.select(WeldManager.class).get();
+            Collection<Class<? extends Annotation>> scopes = wm.getScopes();
+            // test brings no additional scopes we are in SE, hence 7 scopes
+            // Session, request, conversation, application, singleton, thread, dependent
+            Assert.assertEquals(7, scopes.size());
```
[the remark this answers is not in the document]
Comment (manovotn) [reply environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/weldManager/scopes/GetScopesTests.java]: Well, there isn't. The point of the method is to give you access to just that :)
The default is unlikely to change (although if the test were in WFLY, the set would be different).

File: impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java [hunk 2]
```diff
@@ -272,4 +273,13 @@ public BeanManagerImpl unwrap() {
         delegate().disposeAnnotatedType(type, id);
     }
 
+    @Override
+    public boolean isContextActive(Class<? extends Annotation> scopeType) {
```
Comment (nziakova) [inline impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java]: Should this be documented?

File: impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java [hunk 3]
```diff
@@ -272,4 +273,13 @@ public BeanManagerImpl unwrap() {
...
+        return delegate().isContextActive(scopeType);
+    }
+
+    @Override
+    public Collection<Class<? extends Annotation>> getScopes() {
```
Comment (nziakova) [inline impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java]: Should this be documented?

Comment (manovotn) [reply impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java]: Yes, but not quite yet - what needs documenting is Weld API and we are not sure about it's final form yet. But granted, we need a backing JIRA to keep in mind to document all API changes once we settle them - I'll create that.

**Comment by manovotn** [comment]
retest this please

File: impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java (hunk 3 above)
Comment (nziakova) [reply impl/src/main/java/org/jboss/weld/bean/builtin/BeanManagerProxy.java]: Makes sense, thanks.

**Pull request merged**